WooCommerce/Classes/ViewRelated/Orders/OrderListViewController.swift

```diff
@@ -548,23 +546,23 @@ private extension OrderListViewController {
 
 extension OrderListViewController {
     /// Adds ability to select any order
-    /// Used when opening an order with deep link
-    func selectOrder(for orderID: Int64) {
-        // check if already selected
-        guard selectedOrderID != orderID else {
-            return
-        }
-        for identifier in dataSource.snapshot().itemIdentifiers {
+    @discardableResult
+    func selectOrder(for orderID: Int64, canSwitchDetails: Bool) -> Bool {
+        let itemIdentifiers = dataSource.snapshot().itemIdentifiers
+        for identifier in itemIdentifiers {
             if let detailsViewModel = viewModel.detailsViewModel(withID: identifier),
                detailsViewModel.order.orderID == orderID,
                let indexPath = dataSource.indexPath(for: identifier) {
+                if canSwitchDetails && (selectedOrderID != orderID || selectedIndexPath != indexPath) {
```

Comment:
To improve readability, we could explain this logic a little more...

```suggestion
                let orderNotAlreadySelected = selectedOrderID != orderID
                let indexPathNotAlreadySelected = selectedIndexPath != indexPath
                let shouldSwitchDetails = orderNotAlreadySelected || indexPathNotAlreadySelected
                if canSwitchDetails && shouldSwitchDetails {
```

Except I'd perhaps further extract `shouldSwitchDetails` to a helper function – that's just a little tricky to do as a suggestion.

Reply:
Yeah, that is nice and readable!